pkg/svcat/service-catalog/class.go

```diff
@@ -117,11 +125,55 @@ func (sdk *SDK) RetrieveClassByPlan(plan *v1beta1.ClusterServicePlan,
 	return class, nil
 }
 
-// CreateClass returns new created class
-func (sdk *SDK) CreateClass(class *v1beta1.ClusterServiceClass) (*v1beta1.ClusterServiceClass, error) {
-	created, err := sdk.ServiceCatalog().ClusterServiceClasses().Create(class)
+// CreateClassFrom returns new created class
+func (sdk *SDK) CreateClassFrom(opts CreateClassFromOptions) (Class, error) {
+	fromClass, err := sdk.RetrieveClassByName(opts.From)
+	if err != nil {
+		return nil, err
+	}
+
+	if opts.Scope.Matches(ClusterScope) {
+		fromClass.Name = opts.Name
+		return sdk.createClusterServiceClass(fromClass)
+	}
+
+	fromServiceClass := &v1beta1.ServiceClass{
```

Comment:
I think part of the reason that you are having trouble here (having to manually copy over all the fields) because of how you are retrieving the class to copy.

But maybe I'm confused. Let's get the question of how we copy namespaced classes settled, then we can think about how to go about copying based on the type of the source and destination.